x-pack/plugins/snapshot_restore/server/routes/api/snapshots.ts

```diff
@@ -37,6 +37,25 @@ export function registerSnapshotsRoutes({
         // Silently swallow error as policy names aren't required in UI
       }
 
+      let repositories: string[] = [];
+
+      try {
+        const {
+          body: repositoriesByName,
+        } = await clusterClient.asCurrentUser.snapshot.getRepository({
+          repository: '_all',
+        });
+        repositories = Object.keys(repositoriesByName);
+
+        if (repositories.length === 0) {
```

Comment:
It is not possible that you have some snapshot, then delete your repositories but still want to see your snapshots?

Reply:
That is a good question. I don't think it would be possible to still have snapshots if you delete the repositories in which they exist, but let me verify 😄.

For reference, this is how the logic existed previously: https://github.com/elastic/kibana/blob/7.13/x-pack/plugins/snapshot_restore/server/routes/api/snapshots.ts#L54

Reply:
My assumption above is correct. I ran the following commands to verify:

```
// Create repository
PUT /_snapshot/my_repo
{
  "type": "fs",
  "settings": {
    "location": "my_backup_location"
  }
}

// Create 2 snapshots
PUT /_snapshot/my_repo/snapshot_1?wait_for_completion=true
PUT /_snapshot/my_repo/snapshot_2?wait_for_completion=true

// Verify snapshots exist
GET /_snapshot/_all/_all

// Delete repository
DELETE /_snapshot/my_repo

// Verify snapshots do not exist
GET /_snapshot/_all/_all
```

Reply:
Great, thanks for the info that makes sense 👍